From c23b4990379decb5cc59bd53952a534c7f8c8914 Mon Sep 17 00:00:00 2001
From: Sam Bobrowicz <sbobrowicz@digilentinc.com>
Date: Sat, 20 Feb 2016 16:57:26 -0800
Subject: [PATCH 3/9] clk: dglnt-dynclk: Convert parent rate to KHz

Convert the parent rate to KHz before calling find_mode. Fixes a bug that made
the driver non-functional.

Signed-off-by: Sam Bobrowicz <sbobrowicz@digilentinc.com>
---
 drivers/clk/clk-dglnt-dynclk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/clk-dglnt-dynclk.c b/drivers/clk/clk-dglnt-dynclk.c
index b6146105aaa1..924140192b87 100644
--- a/drivers/clk/clk-dglnt-dynclk.c
+++ b/drivers/clk/clk-dglnt-dynclk.c
@@ -408,7 +408,7 @@ static int dglnt_dynclk_set_rate(struct clk_hw *clk_hw,
       return 0;
 
 	rate = (rate + 100) / 200; //Convert from Hz to KHz, then multiply by five to account for BUFR division
-   
+	parent_rate = (parent_rate + 500) / 1000; //convert from Hz to KHz 
    if(!dglnt_dynclk_find_mode(rate, parent_rate, &clkMode))
       return -EINVAL;
 
@@ -429,7 +429,7 @@ static long dglnt_dynclk_round_rate(struct clk_hw *hw, unsigned long rate,
 {
 	struct dglnt_dynclk_mode clkMode;
    
-   dglnt_dynclk_find_mode(((rate + 100) / 200), *parent_rate, &clkMode);
+   dglnt_dynclk_find_mode(((rate + 100) / 200), ((*parent_rate) + 500) / 1000, &clkMode);
 
 	return (clkMode.freq * 200);
 }
-- 
2.17.1

